src/MPI.jl

```diff
@@ -4,21 +4,31 @@ module MPI
 
 using Compat
 
-const depfile = joinpath(dirname(@__FILE__), "..", "deps", "src", "compile-time.jl")
-isfile(depfile) || error("MPI not properly installed. Please run Pkg.build(\"MPI\")")
+@windows_only begin
+	const depfile = "win_mpiconstants.jl"
+end
+
+@unix_only begin
+	const depfile = joinpath(dirname(@__FILE__), "..", "deps", "src", "compile-time.jl")
+	isfile(depfile) || error("MPI not properly installed. Please run Pkg.build(\"MPI\")")
+end
+
 include(depfile)
+
 include("mpi-base.jl")
 include("cman.jl")
 
 function __init__()
-    # need to open libmpi with RTLD_GLOBAL flag for Linux, before any ccall
-    # cannot use RTLD_DEEPBIND; this leads to segfaults at least on Ubuntu 15.10
-    @eval const libmpi_handle =
-        Libdl.dlopen(libmpi, Libdl.RTLD_LAZY | Libdl.RTLD_GLOBAL)
-
-    # look up all symbols ahead of time
-    for (jname, fname) in _mpi_functions
-        eval(:(const $jname = Libdl.dlsym(libmpi_handle, $fname)))
+    @unix_only begin
+        # need to open libmpi with RTLD_GLOBAL flag for Linux, before any ccall
+        # cannot use RTLD_DEEPBIND; this leads to segfaults at least on Ubuntu 15.10
+        @eval const libmpi_handle =
+            Libdl.dlopen(libmpi, Libdl.RTLD_LAZY | Libdl.RTLD_GLOBAL)
+
+        # look up all symbols ahead of time
+        for (jname, fname) in _mpi_functions
+            eval(:(const $jname = Libdl.dlsym(libmpi_handle, $fname)))
```

Comment:
With precompiling, the compiler cannot look up the symbols in the library, since the library used at run time might be different. That, at least, is my understanding of how ELF works. This might be quite irrelevant on Windows.